centreon/www/class/centreonGraph.class.php

```diff
@@ -433,18 +442,18 @@ public function initCurveList()
             /* Create selector for reals metrics */
             if (count($l_rmEnabled)) {
                 $l_rselector =
-                    "metric_id IN (" . implode(
+                     implode(
                         ",",
                         array_map(array("CentreonGraph", "quote"), $l_rmEnabled)
-                    ) . ")";
+                    );
                 $this->log("initCurveList with selector [real]= " . $l_rselector);
             }
             if (count($l_vmEnabled)) {
                 $l_vselector =
-                    "vmetric_id IN (" . implode(
+                    implode(
```

Comment:
This parameter is used to filter on the integer metric_id column, so don't quote integer values and use the preparation statement.